File: dart/dojo/kata9.dart
// https://www.codewars.com/kata/559e3224324a2b6e66000046/train/dart

// 1 <= x <= n, 1 <= y <= n

int sumin(int n) {
  var sum = 0;
  for (var x = 1; x <= n; x++) {
    for (var y = 1; y <= n; y++) {
      sum += x < y ? x : y;
    }
  }
  print(sum);
  return sum;
}

int sumax(int n) {
  var sum = 0;
  for (var x = 1; x <= n; x++) {
    for (var y = 1; y <= n; y++) {
      sum += x > y ? x : y;
    }
  }
  print(sum);
  return sum;
}

int sumsum(int n) {
  var a = sumin(n) + sumax(n);
  print(a);
  return a;
}

// int sumin(int n) => n * (n + 1) * (2 * n + 1) ~/ 6;
// int sumax(int n) => n * (n + 1) * (4 * n - 1) ~/ 6;
// int sumsum(int n) => n * n * (n + 1);

main(List<String> args) {
  assert(sumin(5) == 55);
  // assert(sumax(8) == 372);
  // assert(sumsum(8) == 576);
}

File: kotlin/dojo/kata9.kt
// https://www.codewars.com/kata/54bf1c2cd5b56cc47f0007a1/train/kotlin

// fun duplicateCount(text: String): Int {
//     println(text)
//     // aabbcde
//     // [, a, a, b, b, c, d, e, ]
//     // [, a, b, c, d, e]
//     val a = text.split("")
//     println(a)
    
//     val s = mutableSetOf<String>()
//     a.forEach { s.add(it) }
//     println(s)

//     val solution = a.size - s.size - 1
//     println(solution)

//     return solution
// }toUpperCase().

fun duplicateCount(text: String): Int {
    val a = text.toUpperCase().split("").filter {it.length > 0}.groupingBy {it}.eachCount().values.sumBy {
        if(it > 1) 1
        else 0
    }
    println(a)
    return a
}

fun main() {
    val a = "aabbcde"
    duplicateCount(a)
}

// fun duplicateCount(text: String) = text.groupBy(Char::toLowerCase).count { it.value.count() > 1 }